geomesa-convert/geomesa-convert-json/src/main/scala/org/locationtech/geomesa/convert/json/GeoJsonParsing.scala

```diff
@@ -0,0 +1,71 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.convert.json
+
+import com.google.gson.JsonElement
+import com.vividsolutions.jts.geom._
+import com.vividsolutions.jts.geom.impl.CoordinateArraySequence
+import org.locationtech.geomesa.utils.text.WKTUtils
+
+
+trait GeoJsonParsing {
+
+  import scala.collection.JavaConverters._
+
+  private val CoordsPath = "coordinates"
+
+  private val geoFac = new GeometryFactory
+
+  def toPointCoords(el: JsonElement): Coordinate = {
+    val Seq(x, y) = el.getAsJsonArray.iterator.asScala.map(_.getAsDouble).toSeq
+    new Coordinate(x, y)
+  }
+
+  def toCoordSeq(el: JsonElement): CoordinateSequence =
+    new CoordinateArraySequence(el.getAsJsonArray.iterator.asScala.map(_.getAsJsonArray).map(toPointCoords).toArray)
+
+  def toPolygon(el: JsonElement): Polygon = {
+    val rings = el.getAsJsonArray.iterator.asScala.map(c => geoFac.createLinearRing(toCoordSeq(c)))
+    val shell = rings.next
+    if (rings.hasNext) {
+      geoFac.createPolygon(shell, rings.toArray)
+    } else {
+      geoFac.createPolygon(shell)
+    }
+  }
+
+  def parseGeometry(el: JsonElement): Geometry = {
+    if (el.isJsonObject) {
+      val geomType = el.getAsJsonObject.get("type").getAsString.toLowerCase
+      geomType match {
+        case "point" =>
+          geoFac.createPoint(toPointCoords(el.getAsJsonObject.get(CoordsPath)))
+        case "linestring" =>
+          geoFac.createLineString(toCoordSeq(el.getAsJsonObject.get(CoordsPath)))
+        case "polygon" =>
+          toPolygon(el.getAsJsonObject.get(CoordsPath))
+        case "multipoint" =>
+          geoFac.createMultiPoint(toCoordSeq(el.getAsJsonObject.get(CoordsPath)))
+        case "multilinestring" =>
+          val coords = el.getAsJsonObject.get(CoordsPath).getAsJsonArray
+              .iterator.asScala.map(c => geoFac.createLineString(toCoordSeq(c))).toArray
+          geoFac.createMultiLineString(coords)
+        case "multipolygon" =>
```

Comment:
Can GeoJson have GeometryCollections?

Reply:
indeed, it appears so: https://tools.ietf.org/html/rfc7946#page-9

Reply:
i'll add a follow on ticket

Reply:
https://geomesa.atlassian.net/browse/GEOMESA-2325